File: VB/Step 4 - building views/Helpers/PersonsViewModelCreator.vb
Imports DX_MVVM.Models
Imports DX_MVVM.ViewModels

Namespace DX_MVVM.Helpers
    Public NotInheritable Class PersonsViewModelCreator

        Private Sub New()
        End Sub
        Public Shared ReadOnly PersonsViewModel As PersonsViewModel

        Shared Sub New()
            PersonsViewModel = New PersonsViewModel(PersonsCreator.Persons, AddressOf CreatingMethod)
        End Sub
        Private Shared Function CreatingMethod(ByVal person As Person) As PersonViewModel
            For Each personViewModel As PersonViewModel In PersonViewModelCreator.PersonViewModel
                If personViewModel.Person Is person Then
                    Return personViewModel
                End If
            Next personViewModel
            Return New PersonViewModel(person)
        End Function
    End Class
End Namespace

File: CS/Step 3 - creating view 2 (PersonsViewModel and PersonsView)/Helpers/PersonsViewModelCreator.cs
using DX_MVVM.Models;
using DX_MVVM.ViewModels;

namespace DX_MVVM.Helpers {
    public static class PersonsViewModelCreator {
        public static readonly PersonsViewModel PersonsViewModel;

        static PersonsViewModelCreator() {
            PersonsViewModel = new PersonsViewModel(PersonsCreator.Persons, CreatingMethod);
        }
        static PersonViewModel CreatingMethod(Person person) {
            foreach(PersonViewModel personViewModel in PersonViewModelCreator.PersonViewModel)
                if(personViewModel.Person == person)
                    return personViewModel;
            return new PersonViewModel(person);
        }
    }
}
